libr/core/core.c

```diff
@@ -2713,8 +2713,8 @@ R_API bool r_core_init(RCore *core) {
 	core->printidx = 0;
 	core->lastcmd = NULL;
 	core->cmdlog = NULL;
-	core->charset = r_charset_new ();
-	core->charset->db = sdb_ns (core->sdb, "charset", 1);
+	core->print->charset = r_charset_new ();
+	core->print->charset->db = sdb_ns (core->sdb, "charset", 1);
```

Comment:
as long as core is needed here to set the NS, i think its better to:

* in r_print_new() call print->charset->db = sdb_new0();
* in core_init() after calling r_print_new() use `sdb_ns_set (core->sdb, "charset", core->print->charset->db);`